From 0968e834a268d5fd516944d0a9ada3b90ace9e02 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Thu, 9 May 2019 10:21:03 +0700
Subject: [PATCH 19/23] arm64: dts: renesas: r8a774c0-cat874: enable Display
 Unit

This patch enable DU (Display Unit) in CAT874 Board.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts b/arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts
index 9f1fddd..3923761 100644
--- a/arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts
@@ -120,6 +120,13 @@
 		compatible = "ti,pcm5102a";
 		#sound-dai-cells = <0>;
 	};
+
+	/* External DU dot clocks */
+	x13_clk: x13 {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <74250000>;
+	};
 };
 
 &audio_clk_a {
@@ -205,6 +212,12 @@
 		groups = "tpu0_to0", "tpu0_to1";
 		function = "tpu0";
 	};
+
+	du_pins: du {
+		groups = "du_rgb888", "du_sync", "du_clk_out_0",
+			"du_disp", "du_clk_in_0";
+		function = "du";
+	};
 };
 
 &scif2 {
@@ -462,3 +475,14 @@
 	pinctrl-names = "default";
 	status = "okay";
 };
+
+&du {
+	pinctrl-0 = <&du_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	clocks = <&cpg CPG_MOD 724>,
+		 <&cpg CPG_MOD 723>,
+		 <&x13_clk>;
+	clock-names = "du.0", "du.1", "dclkin.0";
+};
-- 
2.7.4

